src/App.js

```diff
@@ -1,24 +1,92 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState } from 'react';
 import './App.css';
+import Contacts from './contacts';
+
+function ShowContacts() {
+  const [contacts, setContacts] = useState(Contacts.splice(0, 5));
+
+  const randomContactHandler = () => {
+    const contactCopy = [...contacts];
+    const getRandomContact = () =>
+      Contacts[
+        Math.floor(
+          Math.random() * (Contacts.length - (contactCopy.length - 1)) +
+            (contactCopy.length - 1)
+        )
+      ];
+    contactCopy.push(getRandomContact());
+
+    setContacts(contactCopy);
+  };
+
+  const sortNameHandler = () => {
+    let contactCopy = [...contacts];
+
+    contactCopy = contactCopy.sort(function (a, b) {
+      if (a.name < b.name) {
+        return -1;
+      }
+      if (a.name > b.name) {
+        return 1;
+      }
+      return 0;
+    });
+
+    setContacts(contactCopy);
+  };
+
+  const sortPopularityHandler = () => {
+    let contactCopy = [...contacts];
+
+    contactCopy = contactCopy.sort(function (a, b) {
+      return b.popularity - a.popularity;
+    });
+    setContacts(contactCopy);
+  };
+
+  const deleteHandler = (contactId) => {
```

Comment:
Great stuff!
I have one question: if you delete a contact, should it be possible to add it randomly later?
If so, the deleted contact should be added to Contacts.

Reply:
Thanks for the feedback Thomas! There were indeed some things off, I think I resolved them now (or at least made it better)